### DISCUSSION TITLE
Would you like to publish in JSR?

### DISCUSSION THREAD

**Discussion by yumafuu** [wellwelwel/poku#565] [Q&A] [answered]
Hi!

When using `npm:` prefix packages in Deno, compatibility issues can cause anxiety and cause us to not adopt them in a production environment!

I would be very happy if you could publish it to jsr where deno is developed!

**Comment by wellwelwel** [comment]
Hi, @YumaFuu. I can take a look at it. 

I need to understand how interoperability works when the package is published in **JSR** and how it "reacts" to `node:module`.

**Comment by yumafuu** [comment]
Thanks for reply!

I'm not that familiar with it, but it should not be the case that what works with npm will not work with jsr.

You can easily publish from here, so you might want to give it a try.
https://jsr.io/new

**Comment by wellwelwel** [comment]
<blockquote>

### Notes to self

```json
{
  "imports": {
    "node:child_process": "npm:child_process"
  }
}
```

- https://jsr.io/docs/npm-compatibility
- https://docs.deno.com/runtime/manual/node/node_specifiers/
- The project name would be different (e.g, `@wellwelwel/poku`)

</blockquote>

> [!warning]
> 
> The major challenge would be that all the documentation would no longer be universal and there would have to be an exclusive section for **Deno** in each approach.

**Comment by wellwelwel** [comment, marked as answer]
We can use dependencies without installing on **Deno**, a simple approach would be to use **Poku** with `npm:poku` or **esm.sh**, for example:


```ts
import { assert } from 'npm:poku';

assert(true, '🐷');
```

```ts
import { assert } from 'https://esm.sh/poku';

assert(true, '🐷');
```

Also, we can running `poku` command with `npm:poku` without adding it to dependencies:

```sh
deno run --allow-read --allow-env --allow-run npm:poku
```

---

So far, using the **JSR** isn't viable. Let's see the **JSR**'s future:

> <img width="597" alt="Screenshot 2024-07-21 at 10 00 14" src="https://github.com/user-attachments/assets/c39a4f49-fe8c-402d-89d8-5e4e30fc7d57">
>
> - https://twitter.com/JLarky/status/1811100395806363999

---

@YumaFuu, I'll add this discussion to the documentation. Thank you for your contribution 🤝
